nexus/src/app/rack.rs

```diff
@@ -458,11 +459,8 @@ impl super::Nexus {
                     addresses: HashMap::new(),
                 };
 
-                let uplink_address = IpNet::from_str(&format!("{}/32", uplink_config.uplink_ip))
-                    .map_err(|e| Error::internal_error(&format!(
-                    "failed to parse value provided for `rack_network_config.uplink_ip` as `IpNet`: {e}"
-                )))?;
-
+                let uplink_address =
+                    IpNet::V4(Ipv4Net(uplink_config.uplink_cidr));
```

Comment:
This is the first place that warrants an extra look: previously we were converting an IP to an `IP/32`; we now take the `IP/prefix` as-is from RSS. Will this still be correct even if the prefix is something other than 32?

Reply:
For IPv4, prefix > 32 should be an error. For IPv6, prefix > 128 should be an error. The absence of a prefix is generally interpreted as /32 for IPv4 and /128 for IPv6. As long as these conventions hold, we should be good.

Reply:
:+1: `Ipv4Network` (the type of `uplink_config.uplink_cidr` here) ensures that the prefix is <= 32.